src/classes/CRLP_RollupCMT.cls

```diff
@@ -408,6 +462,127 @@ public class CRLP_RollupCMT {
+        * @description Return the rollup type for a given rollup definition
+        * @param Rollup__mdt CMT Record
+        * @return CRLP_RollupProcessingOptions.RollupType
+        **************************************************************************************************************/
+        public CRLP_RollupProcessingOptions.RollupType getRollupType() {
+
+            String accountApiName = UTIL_Describe.getObjectDescribe('Account').getName();
+            String contactApiName = UTIL_Describe.getObjectDescribe('Contact').getName();
+            String opportunityApiName = UTIL_Describe.getObjectDescribe('Opportunity').getName();
+            String paymentApiName = UTIL_Describe.getObjectDescribe('npe01__OppPayment__c').getName();
+            String recurringDonationsApiName = UTIL_Describe.getObjectDescribe('npe03__Recurring_Donation__c').getName();
+            String allocationApiName = UTIL_Describe.getObjectDescribe(UTIL_Namespace.StrAllNSPrefix('Allocation__c')).getName();
+            String partialSoftCreditApiName = UTIL_Describe.getObjectDescribe(UTIL_Namespace.StrAllNSPrefix('Partial_Soft_Credit__c')).getName();
+            String accountSoftCreditApiName = UTIL_Describe.getObjectDescribe(UTIL_Namespace.StrAllNSPrefix('Account_Soft_Credit__c')).getName();
+
+            // work down from most specific
+            if (this.summaryObject == recurringDonationsApiName) {
+                return CRLP_RollupProcessingOptions.RollupType.RecurringDonations;
+            }
+
+            if (this.summaryObject == allocationApiName) {
+                return CRLP_RollupProcessingOptions.RollupType.GAU;
+            }
+
+            if (this.amountObject == partialSoftCreditApiName) {
+                if (this.summaryObject == accountApiName) {
+                    return CRLP_RollupProcessingOptions.RollupType.AccountContactSoftCredit;
+                }
+                if (this.summaryObject == contactApiName) {
+                    return CRLP_RollupProcessingOptions.RollupType.ContactSoftCredit;
+                }
+            }
+
+            if (this.amountObject == accountSoftCreditApiName) {
+                if (this.detailObject == opportunityApiName || this.detailObject == accountSoftCreditApiName) {
+                    return CRLP_RollupProcessingOptions.RollupType.AccountSoftCredit;
+                }
+            }
+
+            if (this.detailObject == opportunityApiName || this.detailObject == paymentApiName) {
+                if (this.summaryObject == accountApiName) {
+                    return CRLP_RollupProcessingOptions.RollupType.AccountHardCredit;
+                }
+                if (this.summaryObject == contactApiName) {
+                    return CRLP_RollupProcessingOptions.RollupType.ContactHardCredit;
+                }
+            }
+
+            return null;
+        }
```

Comment:
I was looking for something that provided me this information but it doesn't look like there was one.  I took the logic from a couple of spots - this needs to be reviewed to make sure it is correct.

Reply:
Other than my suggestions above, this is good. There was no single method that determined this previously.